tutorbackup/templates/backup/build/backup/backup_services.py

```diff
@@ -44,15 +44,33 @@ def mysqldump():
     host = ENV['MYSQL_HOST']
     port = ENV['MYSQL_PORT']
     outfile = MYSQL_DUMPFILE
+    dblist = []
 
     mysql_databases = ENV.get('MYSQL_DATABASES')
     if mysql_databases:
         databases_statement = f"--databases {mysql_databases}"
         logger.info(f"Dumping MySQL databases {mysql_databases} "
                     f"on {host}:{port} to {outfile}")
     else:
-        databases_statement = "--all-databases"
-        logger.info(f"Dumping all MySQL databases "
+        sql_query="""SELECT schema_name FROM information_schema.schemata WHERE schema_name NOT IN
+             ('mysql', 'sys', 'performance_schema', 'information_schema')"""
+
+        mysql_cmd = (f"mysql --host={host} --port={port} --user={user} --password={password} -BANe \"{sql_query}\" ")
+
+        logger.info("Retrieving the list of database schemas to backup")
+        try:
+            dbout = check_output([mysql_cmd], stderr=STDOUT, timeout=5, shell=True, universal_newlines=True)
+        except CalledProcessError as e:
+            logger.error("MySQL Command Status : FAIL", e.returncode, e.output)
+        else:
+            # The element at position [0] is the following message, so we have to exclude it from the schemas list:
+            # "mysql: [Warning] Using a password on the command line interface can be insecure."
+            dblist = dbout.splitlines()[1:]
```

Comment:
Actually, when I tried this locally (I cannot run the plugin with my changes directly so I just experiment with a local script) I tried omitting the `stderr=STDOUT` and I got nothing in return.
My guess is that the query output is following the mysql's non-critical warning (about using a password on the command line) as a single output altogether and so without the `stderr=STDOUT` it cannot be retrieved.
I might be mistaken, but that's how it worked for me when I tried it.

If I don't use the `check_output` though as you have suggested, but the `Popen` instead, I could indeed get just the query output without the warning, like so:

```
dbout = Popen(mysql_cmd, stdout=PIPE, stderr=PIPE, shell=True, universal_newlines=True)
with dbout as proc:
        dblist = proc.stdout.read().strip().split("\n")
```

(The `strip()` on the last line is to get rid of the trailing new line character and the `split("\n")` splits the output into a list, because in the output the db names are separated with the new line character, like so: `'openedx\nora\nreports\n ...'` )

For me both ways seem ok, but I can change it to whichever you think will work most efficiently, with or without the try/except as well. Or if you have another suggestion I can also try it.

---
Thank you for the suggestions/corrections on my code, I know it's not perfect, especially since I cannot try it on the specific plugin directly, so I really appreciate your time helping me with it.